lib/segment/src/vector_storage/chunked_mmap_vectors.rs

```diff
@@ -0,0 +1,195 @@
+use std::fs::{create_dir_all, OpenOptions};
+use std::io::Write;
+use std::marker::PhantomData;
+use std::mem::size_of;
+use std::path::{Path, PathBuf};
+
+use memmap2::MmapMut;
+use serde::{Deserialize, Serialize};
+
+use crate::common::mmap_ops::{
+    open_write_mmap, transmute_from_u8_to_array, transmute_to_u8, transmute_to_u8_array,
+};
+use crate::data_types::vectors::VectorElementType;
+use crate::entry::entry_point::{OperationError, OperationResult};
+
+#[cfg(test)]
+const DEFAULT_CHUNK_SIZE: usize = 4 * 1024 * 1024; // 4Mb
+
+#[cfg(not(test))]
+const DEFAULT_CHUNK_SIZE: usize = 128 * 1024 * 1024; // 128Mb
+
+const CONFIG_FILE_NAME: &str = "config.json";
+const STATUS_FILE_NAME: &str = "status.json";
+
+const MMAP_CHUNKS_PATTERN_START: &str = "chunk_";
+const MMAP_CHUNKS_PATTERN_END: &str = ".mmap";
+
+struct ChunkedMMapStatus {
+    len: usize,
+}
+
+#[derive(Serialize, Deserialize)]
+struct ChunkedMMapConfig {
+    chunk_size: usize,
+    dim: usize,
+}
+
+pub struct ChunkedMMapVectors<T> {
```

Comment:
Nitpick: in other places we write mmap styled as `Mmap`. It may be good to be consistent.

Reply:
:+1: